server/rpc_test.go

```diff
@@ -532,3 +533,104 @@ func defaultTicket(t *testing.T) *net.Ticket {
 		RecipientRandHash: pm.RandBytes(123),
 	}
 }
+
+func StubBroadcastSession(transcoder string) *BroadcastSession {
+	return &BroadcastSession{
+		Broadcaster:      StubBroadcaster2(),
+		ManifestID:       core.RandomManifestID(),
+		OrchestratorInfo: &net.OrchestratorInfo{Transcoder: transcoder},
+	}
+}
+
+func StubBroadcastSessionsManager() *BroadcastSessionsManager {
+	sess1 := StubBroadcastSession("transcoder1")
+	sess2 := StubBroadcastSession("transcoder2")
+
+	return &BroadcastSessionsManager{
+		broadcastSessions: []*BroadcastSession{sess1, sess2},
+		broadcastSessMap: map[string]*BroadcastSession{
+			sess1.OrchestratorInfo.Transcoder: sess1,
+			sess2.OrchestratorInfo.Transcoder: sess2,
+		},
+		sessLock: &sync.Mutex{},
+	}
+}
+
+func TestSelect(t *testing.T) {
+	bsm := StubBroadcastSessionsManager()
+	expectedSess1 := bsm.broadcastSessions[1]
+	expectedSess2 := bsm.broadcastSessions[0]
+
+	// assert that initial lengths are as expected
+	assert := assert.New(t)
+	assert.Len(bsm.broadcastSessions, 2)
+
+	// assert last session selected and broadcastSessions is correct length
+	sess := bsm.selectSession()
+	assert.Equal(expectedSess1, sess)
+	assert.Len(bsm.broadcastSessions, 1)
+
+	sess = bsm.selectSession()
+	assert.Equal(expectedSess2, sess)
+	assert.Len(bsm.broadcastSessions, 0)
+
+	// assert no session is selected from empty list
+	sess = bsm.selectSession()
+	assert.Nil(sess)
+	assert.Len(bsm.broadcastSessions, 0)
+}
+
+func TestRemoveSession(t *testing.T) {
+	bsm := StubBroadcastSessionsManager()
+	sess1 := bsm.broadcastSessions[0]
+	sess2 := bsm.broadcastSessions[1]
+
+	assert := assert.New(t)
+	assert.Len(bsm.broadcastSessMap, 2)
+
+	bsm.removeSession(sess1)
+	assert.Len(bsm.broadcastSessMap, 1)
+
+	bsm.removeSession(sess1)
```

Comment:
Looking great -- maybe a comments along the lines of `// remove nonexistent session` would help here. Also it might generally be good to do pre/post checks that the sessions are actually in the map, eg

```
assert.NotNil(bsm.broadcastSessMap[sess])
bsm.removeSession(sess1)
assert.Nil(bsm.broadcastSessMap[sess])
```